File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.helper(nums, target, 0, len(nums) - 1)

    def helper(self, nums, target, left, right):
        if left > right:
            return -1
        mid = (left + right) // 2
        if nums[mid] == target:
            return mid
        if nums[mid] < target:
            return self.helper(nums, target, mid + 1, right)
        else:
            return self.helper(nums, target, left, mid - 1)

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var left = 0
        var right = numbers.size - 1

        while (left < right) {
            val lower = numbers[left]
            val higher = numbers[right]
            val sum = lower + higher
            when {
                sum == target -> return intArrayOf(left + 1, right + 1)
                sum < target -> left++
                sum > target -> right--
            }
        }
        return intArrayOf(-1, -1)
    }
}
